Lib/test/test_typing.py

```diff
@@ -7104,8 +7104,8 @@ def test_optional_keys(self):
         class Point2Dor3D(Point2D, total=False):
             z: int
 
-        assert Point2Dor3D.__required_keys__ == frozenset(['x', 'y'])
-        assert Point2Dor3D.__optional_keys__ == frozenset(['z'])
+        self.assertEqual(Point2Dor3D.__required_keys__, frozenset(['x', 'y']))
+        self.assertEqual(Point2Dor3D.__optional_keys__, frozenset(['z']))
```

Comment:
The changes in this PR look good. But it looks like we have a lot of tests in this test class that do things like `self.assertEqual(some_frozenset, frozenset(['x', 'y']))`. That _looks_ like it's checking that `some_frozenset` is a frozenset, but it's actually not, because frozensets and sets with the same members compare equal. So I wonder if we should be doing this:

```suggestion
        self.assertEqual(Point2Dor3D.__required_keys__, {'x', 'y'})
        self.assertIsInstance(Point2Dor3D.__required_keys__, frozenset)
        self.assertEqual(Point2Dor3D.__optional_keys__, {'z'})
        self.assertIsInstance(Point2Dor3D.__optional_keys__, frozenset)
```

But maybe we don't need to assert the type of `__required_keys__` every time we want to check the contents of the set.

Reply:
Anyway, we don't need to change that in this PR; the changes here are definitely good.

Reply:
Might as well do that in this PR too.

Reply:
Pushed a few new asserts checking that it's really a frozenset